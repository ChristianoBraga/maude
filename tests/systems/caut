#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < caut.maude -no-banner -no-advise \
  > caut.out 2>&1

diff caut.expected caut.out > /dev/null 2>&1
