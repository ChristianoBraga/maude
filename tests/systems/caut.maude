set show timing off .

load ../../contrib/systems/caut.maude

mod 3x3GRID is
    including PERIODIC-GRID .

    vars S L U R D : State . var SS : States .

    --- calculate the next state as a function of the neighborhood

   ops X - : -> State .
   --------------------

    eq width  = 3 .
    eq height = 3 .

    eq - -> up[-] down[D] left[-] right[-] = - -> X .
    eq S -> up[U] down[-] left[-] right[X] = S -> X .
    eq S -> up[X] down[X] left[L] right[R] = S -> X .
    eq S -> up[U] down[D] left[L] right[R] = S -> - [owise] .

    var N : Nat .

    op init : Nat -> Cells .
    ------------------------
    eq init(0)
     = < 0 , 0 > :: -   ;   < 0 , 1 > :: X   ;   < 0 , 2 > :: -
     ; < 1 , 0 > :: X   ;   < 1 , 1 > :: -   ;   < 1 , 2 > :: -
     ; < 2 , 0 > :: X   ;   < 2 , 1 > :: X   ;   < 2 , 2 > :: X .
endm

reduce [ init(0) ] .
reduce { init(0) } .

--- Show repeating cycle of 3 states from initial board
search [ init(0) ] =>* { CS:Cells } .
