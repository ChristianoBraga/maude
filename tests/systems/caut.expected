==========================================
reduce in 3x3GRID : [init(0)] .
rewrites: 451
result Conf: [< 0,0 > :: - -> X
; < 0,1 > :: X -> -
; < 0,2 > :: - -> -
; < 1,0 > :: X -> -
; < 1,1 > :: - -> X
; < 1,2 > :: - -> -
; < 2,0 > :: X -> -
; < 2,1 > :: X -> -
; < 2,2 > :: X -> -]
==========================================
reduce in 3x3GRID : {init(0)} .
rewrites: 55
result Conf: {< 0,0 > :: -
; < 0,1 > :: X
; < 0,2 > :: -
; < 1,0 > :: X
; < 1,1 > :: -
; < 1,2 > :: -
; < 2,0 > :: X
; < 2,1 > :: X
; < 2,2 > :: X}
==========================================
search in 3x3GRID : [init(0)] =>* {CS:Cells} .

Solution 1 (state 1)
states: 2  rewrites: 461
CS:Cells --> < 0,0 > :: X
; < 0,1 > :: -
; < 0,2 > :: -
; < 1,0 > :: -
; < 1,1 > :: X
; < 1,2 > :: -
; < 2,0 > :: -
; < 2,1 > :: -
; < 2,2 > :: -

Solution 2 (state 3)
states: 4  rewrites: 868
CS:Cells --> < 0,0 > :: -
; < 0,1 > :: -
; < 0,2 > :: X
; < 1,0 > :: X
; < 1,1 > :: -
; < 1,2 > :: -
; < 2,0 > :: X
; < 2,1 > :: -
; < 2,2 > :: X

Solution 3 (state 5)
states: 6  rewrites: 1275
CS:Cells --> < 0,0 > :: X
; < 0,1 > :: X
; < 0,2 > :: -
; < 1,0 > :: -
; < 1,1 > :: -
; < 1,2 > :: X
; < 2,0 > :: -
; < 2,1 > :: -
; < 2,2 > :: -

Solution 4 (state 7)
states: 8  rewrites: 1682
CS:Cells --> < 0,0 > :: X
; < 0,1 > :: -
; < 0,2 > :: -
; < 1,0 > :: -
; < 1,1 > :: X
; < 1,2 > :: -
; < 2,0 > :: X
; < 2,1 > :: X
; < 2,2 > :: -

Solution 5 (state 9)
states: 10  rewrites: 2089
CS:Cells --> < 0,0 > :: -
; < 0,1 > :: X
; < 0,2 > :: X
; < 1,0 > :: X
; < 1,1 > :: -
; < 1,2 > :: -
; < 2,0 > :: -
; < 2,1 > :: -
; < 2,2 > :: -

Solution 6 (state 11)
states: 12  rewrites: 2496
CS:Cells --> < 0,0 > :: -
; < 0,1 > :: X
; < 0,2 > :: -
; < 1,0 > :: -
; < 1,1 > :: -
; < 1,2 > :: X
; < 2,0 > :: -
; < 2,1 > :: X
; < 2,2 > :: X

Solution 7 (state 13)
states: 14  rewrites: 2903
CS:Cells --> < 0,0 > :: X
; < 0,1 > :: -
; < 0,2 > :: X
; < 1,0 > :: -
; < 1,1 > :: X
; < 1,2 > :: -
; < 2,0 > :: -
; < 2,1 > :: -
; < 2,2 > :: -

No more solutions.
states: 15  rewrites: 3310
Bye.
